pkg/detector/detector.go

```diff
@@ -922,35 +919,12 @@ func (d *ResourceDetector) OnPropagationPolicyUpdate(oldObj, newObj interface{})
 	// The idea of the long-term solution, perhaps PropagationPolicy could have
 	// a status, in that case we can record the observed priority(.status.observedPriority)
 	// which can be used to detect priority changes during reconcile logic.
+	policyObj := newObj.(*policyv1alpha1.PropagationPolicy)
```

Comment:
This line should be moved into the `if features.FeatureGate.Enabled(features.PolicyPreemption)` if block, I don't see it used somewhere outside of the if block.

Reply:
done